packages/nextjs-mf/src/plugins/LoadDependenciesTemplate.ts

```diff
@@ -0,0 +1,45 @@
+//@ts-ignore
+import Template from '../../utils/Template';
+
+export const loadDependenciesTemplate = (RuntimeGlobals: any) => {
+  return Template.asString([
+    'function loadDependencies(libKeys, cnn) {',
+    "console.log('loadDependencies', libKeys, cnn);",
+    Template.indent([
+      'libKeys.map(function (libKey) {',
+      Template.indent([
+        // Extract the share key and share version from the libKey array.
+        'var shareKey = libKey[0];',
+        'var shareVersion = libKey[1];',
+
+        // Get the default share scope map.
+        `var shareScopeMap = ${RuntimeGlobals.shareScopeMap}.default;`,
+
+        // Determine the alternative version of the shared module.
+        'var alternativeVersion = Object.keys(shareScopeMap[shareKey])[0];',
+
+        // Get the shared module based on the preferred version or the alternative version.
+        'var lib = (preferredModules.includes(shareKey) && shareScopeMap[shareKey][shareVersion]) ? shareScopeMap[shareKey][shareVersion] : shareScopeMap[shareKey][alternativeVersion];',
+        // 'console.info("lib", {lib, shareKey, preferredModules, shareVersion,shareScopeMap });',
+        Template.indent([
+          'if (!lib.loaded) {',
+          Template.indent([
+            'lib.loaded = 1;',
+            "console.log('loading', shareKey, shareVersion);",
+            'asyncQueue.push(lib.get().then(function (f) {',
```

Comment:
Push get call into async queue, upon get resolve, replace self reference with resolved output. 

When webpack entrypoint calls webpack/sharing/consumes/react, webpack runtime calls shareScope.get() which now returns the require factory, not a promise.